search/engine.go

```diff
@@ -418,30 +515,49 @@ func (e *ESEngine) DoSearch(ctx context.Context, query Query, sortBy string, fro
 		return nil, errors.Wrap(err, "ES error.")
 	}
 
-	if len(mr.Responses) != 2*len(query.LanguageOrder) {
+	if len(mr.Responses) != indexesCount*len(query.LanguageOrder) {
 		return nil, errors.New(fmt.Sprintf("Unexpected number of results %d, expected %d",
-			len(mr.Responses), 2*len(query.LanguageOrder)))
+			len(mr.Responses), indexesCount*len(query.LanguageOrder)))
 	}
 
 	// Interleave content units and collection results by language.
 	// Then go over responses and choose first not empty retults list.
 	for i := 0; i < len(query.LanguageOrder); i++ {
-		cuR := mr.Responses[i]
-		cR := mr.Responses[i+len(query.LanguageOrder)]
-		if cuR.Error != nil {
-			log.Warnf("%+v", cuR.Error)
-			return nil, errors.New("Failed multi get.")
+
+		var cuR *elastic.SearchResult
+		var cR *elastic.SearchResult
+		var sR *elastic.SearchResult
+		var joinedR *elastic.SearchResult
+
+		if indexesCount >= 2 {
+			cuR = mr.Responses[i]
+			if cuR.Error != nil {
+				log.Warnf("%+v", cuR.Error)
+				return nil, errors.New("Failed multi get.")
+			}
+			cR = mr.Responses[i+len(query.LanguageOrder)]
+			if cR.Error != nil {
+				log.Warnf("%+v", cR.Error)
+				return nil, errors.New("Failed multi get.")
+			}
 		}
-		if cR.Error != nil {
-			log.Warnf("%+v", cR.Error)
-			return nil, errors.New("Failed multi get.")
+
+		if indexesCount == 1 || indexesCount == 3 {
+			sR = mr.Responses[i+(len(query.LanguageOrder)*(indexesCount-1))]
+			if sR.Error != nil {
+				log.Warnf("%+v", sR.Error)
+				return nil, errors.New("Failed multi get.")
+			}
 		}
+
 		if haveHits(cuR) || haveHits(cR) {
-			log.Debugf("Joining:\n%+v\n\n%+v\n\n\n", cuR.Hits, cR.Hits)
-			ret, err := joinResponses(cuR, cR, sortBy, from, size)
-			log.Debugf("Res: %+v", ret.Hits)
-			return ret, err
+			joinedR, err = joinResponses(cuR, cR, sortBy, from, size)
+			if err != nil {
+				return joinedR, err
+			}
 		}
+		joinedR, err = joinResponses(joinedR, sR, sortBy, from, size)
```

Comment:
Rename `joinReponses` to `joinTwoResponses`
Implement new joinResponses function as variadic:
```
func joinResponses(sortBy string, from int, size int, results... *elastic.SearchResult)) {
 ...
   use somewhere here joinTwoResponses and comment to refactor it later.
 ...
}
Within `joinReponses` `results` will be []*elastic.SearchResult{cR, cuR, sR}

joinedR, err = joinResponses(sortBy, from, size, cuR, cR, sR)
```

Reply:
Issue created AS-36